bibtex/generate_bibtex.py

```diff
@@ -0,0 +1,100 @@
+import os
+import urllib.request
+import requests
+import re
+
+def download_and_write(filepath,outfile="readme.md"):
+    """
+    If we want to download upstream
+    """
+    # Create a file object to write the downloaded file to.
+    with open(outfile, 'wb') as f:
+        response = requests.get(fileurl, stream=True)
+        for chunk in response.iter_content(chunk_size=1024):
+            f.write(chunk)
+                                                                                                                    
+def extract_dois(filepath,url=False):
+    if url:
+        outfile = download_and_write(filepath)
+    else:
+        outfile = filepath
+        
+    # Regular expression pattern for DOIs
+    doi_pattern = r'\[.*?\]\((https?://doi\.org/[-._;()/:a-zA-Z0-9]+)\)'
```

Comment:
Often, I use `https://dx.doi.org/...`. A long time ago, I remember reading something (I think from the official website) about this being better. Couldn't find it after a quick search. Do you mind updating this so that the dx.doi is an allowable pattern, without opening it to other patterns like `https://<some other text>.doi.org/...`? See https://regex101.com/r/8Muini/3

Aside: I came across a few similar regex's that don't assume the DOI prefix. However, these are for DOIs rather than DOIs encoded in URLs. More of a provenance thing mentioning them here.
- https://www.crossref.org/blog/dois-and-matching-regular-expressions/
- https://stackoverflow.com/questions/27910/finding-a-doi-in-a-document-or-page

Reply:
I didn't catch this, thanks for pointing it out. The following regex should handle this (tested in your regex101):

`\[.*?\]\((https?://(dx\.)?doi\.org/[-._;()/:a-zA-Z0-9]+)\)`

will add.

Reply:
implemented in commit [5718c69](https://github.com/sgbaird/awesome-self-driving-labs/pull/10/commits/5718c69d1889a5ee93cbc78fbde73327a85aa30d).